# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/credentialmanagement/identity/NavigatorLogin.h
# Commit: 0e3bc029e59f
# Full Hash: 0e3bc029e59f5f0319a57c469d7c0991081379ff
# Author: Benjamin VanderSloot <bvandersloot@mozilla.com>
# Date: 2025-06-10 21:43:01
# Regressor Bug: 1969017
# File Overlap Count: 2
# Description:
#   Bug 1969017 - Refactor FedCM to use a managed actor rather than tagging onto WindowGlobalParent - r=anti-tracking-reviewers,emz
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D251617
# ==============================================================================

diff -r 8b2ffdf05455 -r 0e3bc029e59f dom/credentialmanagement/identity/NavigatorLogin.h
--- a/dom/credentialmanagement/identity/NavigatorLogin.h	Tue Jun 10 11:49:57 2025 +0000
+++ b/dom/credentialmanagement/identity/NavigatorLogin.h	Tue Jun 10 11:49:57 2025 +0000
@@ -22,8 +22,8 @@
   NS_INLINE_DECL_CYCLE_COLLECTING_NATIVE_REFCOUNTING(NavigatorLogin)
   NS_DECL_CYCLE_COLLECTION_NATIVE_WRAPPERCACHE_CLASS(NavigatorLogin)
 
-  explicit NavigatorLogin(nsIGlobalObject* aGlobal);
-  nsIGlobalObject* GetParentObject() const { return mOwner; }
+  explicit NavigatorLogin(nsPIDOMWindowInner* aGlobal);
+  nsPIDOMWindowInner* GetParentObject() const { return mOwner; }
   virtual JSObject* WrapObject(JSContext* aCx,
                                JS::Handle<JSObject*> aGivenProto) override;
 
@@ -41,7 +41,7 @@
   virtual ~NavigatorLogin();
 
  private:
-  nsCOMPtr<nsIGlobalObject> mOwner;
+  nsCOMPtr<nsPIDOMWindowInner> mOwner;
 };
 
 }  // namespace mozilla::dom